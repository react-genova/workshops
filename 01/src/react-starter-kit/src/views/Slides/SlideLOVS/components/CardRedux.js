//@flow
import React from 'react';
import { Card, CardElement } from '../../../Card';

type Props = {
    onClick:(url:string)=>void
};

const CardRedux = ({onClick}:Props) => (
    <Card title="Redux" description="Dan Abramov AKA gaeron AKA the genius">
        <CardElement title="redux" description="the core of the business logic" url="https://redux.js.org/" onClick={onClick} />
        <CardElement title=" react-redux" description="link our two main actors" url="https://github.com/reactjs/react-redux" onClick={onClick} />
        <CardElement title="reselect" description="cool selectors management" url="https://github.com/reactjs/reselect" onClick={onClick} />
        <CardElement title="redux-immutable" description="immutable in redux store" url="https://github.com/gajus/redux-immutable" onClick={onClick} />
    </Card>
);

export default CardRedux;