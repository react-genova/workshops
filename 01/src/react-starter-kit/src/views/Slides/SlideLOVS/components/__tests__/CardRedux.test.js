//@flow
import React from 'react';
import { shallow } from 'enzyme';
import CardRedux from '../CardRedux';

describe('Card Redux', () => {

    it('snapshot check', () => {
        const onClick = (url:string)=>{url.toLocaleLowerCase();};
        const component = shallow(<CardRedux onClick={onClick} />);
        expect(component).toMatchSnapshot();
    });

});