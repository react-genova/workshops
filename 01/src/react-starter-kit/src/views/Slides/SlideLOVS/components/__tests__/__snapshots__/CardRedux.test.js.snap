// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Redux snapshot check 1`] = `
<Card
  description="Dan Abramov AKA gaeron AKA the genius"
  title="Redux"
>
  <CardElement
    description="the core of the business logic"
    onClick={[Function]}
    title="redux"
    url="https://redux.js.org/"
  />
  <CardElement
    description="link our two main actors"
    onClick={[Function]}
    title=" react-redux"
    url="https://github.com/reactjs/react-redux"
  />
  <CardElement
    description="cool selectors management"
    onClick={[Function]}
    title="reselect"
    url="https://github.com/reactjs/reselect"
  />
  <CardElement
    description="immutable in redux store"
    onClick={[Function]}
    title="redux-immutable"
    url="https://github.com/gajus/redux-immutable"
  />
</Card>
`;
